eks/deploy_state.go

```diff
@@ -0,0 +1,92 @@
+package eks
+
+import (
+	"github.com/gruntwork-io/kubergrunt/logging"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/util/json"
+	"os"
+)
+
+const defaultStateFile = "./.kubergrunt.state"
+
+type DeployState struct {
+	GatherASGInfoDone      bool
+	SetMaxCapacityDone     bool
+	ScaleUpDone            bool
+	WaitForNodesDone       bool
+	CordonNodesDone        bool
+	DrainNodesDone         bool
+	DetachInstancesDone    bool
+	TerminateInstancesDone bool
+	RestoreCapacityDone    bool
+
+	Path string
+	ASG  ASG
+}
+
+type ASG struct {
+	Name                 string
+	OriginalCapacity     int64
+	MaxCapacityForUpdate int64
+	OriginalMaxCapacity  int64
+	OriginalInstances    []string
+	NewInstances         []string
+}
+
+func readOrInitializeState(file string, ignoreExistingFile bool) (*DeployState, error) {
+	logger := logging.GetProjectLogger()
+	if ignoreExistingFile {
+		logger.Info("Ignore existing state file.")
+		return newDeployState(file), nil
+	}
+
+	logger.Infof("Looking for existing recovery file %s", file)
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		logger.Debugf("No state present, creating new: %s", err.Error())
+		return newDeployState(file), nil
+	}
+	var parsedState DeployState
+	err = json.Unmarshal(data, &parsedState)
+	if err != nil {
+		return nil, err
+	}
+	return &parsedState, nil
+}
+
+func (state *DeployState) persist() {
+	file := state.Path
+	logger := logging.GetProjectLogger()
+	logger.Debugf("storing state file %s", file)
+
+	data, err := json.Marshal(state)
+
+	if err != nil {
+		logger.Fatalf("Error marshaling state: %v", err)
```

Comment:
I think as a general pattern we try to avoid `Fatal` to shutdown the CLI. This is primarily because we would lose the stack trace error reporting we have in the framework ([here](https://github.com/gruntwork-io/go-commons/blob/master/entrypoint/entrypoint.go#L47)). Can you switch this to return the error (with stack trace) and handle it in the caller? I know that increases cyclomatic complexity, but it helps when debugging the errors (e.g., which action failed to persist the state?).